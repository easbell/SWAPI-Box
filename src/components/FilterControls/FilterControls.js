import React, { Component } from 'react'

const FilterControls = ( {handleSort} ) => {
  const handleSelection = (e) => {
    handleSort(e.target.className)
  }

  return (
    <div>
      <button onClick={handleSelection} className='people'>People</button>
      <button onClick={handleSelection} className='planets'>Planets</button>
      <button onClick={handleSelection} className='vehicles'>Vehicles</button>
    </div>
  )
}

export default FilterControls
