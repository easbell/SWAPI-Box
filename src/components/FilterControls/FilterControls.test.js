import React from 'react';
import FilterControls from './FilterControls';
import { mount } from 'enzyme';

describe('FilterControls', () => {
  let wrapper;
  let mockHandleSort;

  beforeEach(() => {
    mockHandleSort = jest.fn()
    wrapper = mount(<FilterControls handleSort={mockHandleSort}/>)
  })
  it('should match snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke handleSort with proper argument when button is clicked', () => {
    wrapper.find('.people').simulate('click');
    expect(mockHandleSort).toHaveBeenCalledWith('people')
  });

  it('should invoke handleSort with proper argument when button is clicked', () => {
    wrapper.find('.planets').simulate('click');
    expect(mockHandleSort).toHaveBeenCalledWith('planets')
  });

  it('should invoke handleSort with proper argument when button is clicked', () => {
    wrapper.find('.vehicles').simulate('click');
    expect(mockHandleSort).toHaveBeenCalledWith('vehicles')
  });
})
