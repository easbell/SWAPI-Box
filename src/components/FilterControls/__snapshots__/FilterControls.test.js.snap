// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterControls should match snapshot with all data passed in 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <FilterControls
    handleSort={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "handleSort": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="filter"
      >
        <button
          class="people"
        >
          People
        </button>
        <button
          class="planets"
        >
          Planets
        </button>
        <button
          class="vehicles"
        >
          Vehicles
        </button>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people"
            onClick={[Function]}
          >
            People
          </button>,
          <button
            className="planets"
            onClick={[Function]}
          >
            Planets
          </button>,
          <button
            className="vehicles"
            onClick={[Function]}
          >
            Vehicles
          </button>,
        ],
        "className": "filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <button
            class="people"
          >
            People
          </button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "People",
          ],
          "type": "button",
        },
        Object {
          "instance": <button
            class="planets"
          >
            Planets
          </button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Planets",
          ],
          "type": "button",
        },
        Object {
          "instance": <button
            class="vehicles"
          >
            Vehicles
          </button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Vehicles",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "handleSort": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="filter"
        >
          <button
            class="people"
          >
            People
          </button>
          <button
            class="planets"
          >
            Planets
          </button>
          <button
            class="vehicles"
          >
            Vehicles
          </button>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="people"
              onClick={[Function]}
            >
              People
            </button>,
            <button
              className="planets"
              onClick={[Function]}
            >
              Planets
            </button>,
            <button
              className="vehicles"
              onClick={[Function]}
            >
              Vehicles
            </button>,
          ],
          "className": "filter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <button
              class="people"
            >
              People
            </button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "className": "people",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "People",
            ],
            "type": "button",
          },
          Object {
            "instance": <button
              class="planets"
            >
              Planets
            </button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "className": "planets",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Planets",
            ],
            "type": "button",
          },
          Object {
            "instance": <button
              class="vehicles"
            >
              Vehicles
            </button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "className": "vehicles",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Vehicles",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
